clusterloader2/testing/nichotr/config.yaml

```diff
@@ -0,0 +1,143 @@
+{{$saturationDeploymentTimeout := DivideFloat $totalPods $DENSITY_TEST_THROUGHPUT | AddInt $MIN_SATURATION_PODS_TIMEOUT}}
+
+# Methods that don't work
+# APResponsiveness
+# CPUProfile
+# ETCD Metrics
+# Memory Profile
+# MetricsForE2E
+# Scheduling Metrics
+# OOM Tracking
+# InClusterNetworkLatency
+# DNSLookupLatency
+
+name: simple bring up
+namespace:
+  number: {{$namespaces}}
+tuningSets:
+- name: Uniform5qps
+  qpsLoad:
+    qps: 100
+
+steps:
+- name: Starting Measurements
+  measurements:
+  - Identifier: ScaleUpTimer
+    Method: Timer
+    Params:
+      action: start
+      label: scale_up_time
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+      labelSelector: group = nichotr
+      threshold: 1h
+  - Identifier: ResourceUsageSummary
+    Method: ResourceUsageSummary
+    Params:
+      action: start
+  - Identifier: SchedulingThroughput
+    Method: SchedulingThroughput
+    Params:
+      action: start
+
+- name: Creating saturation pods
+  phases:
+    - namespaceRange:
+        min: 1
+        max: {{$namespaces}}
+      replicasPerNamespace: 1
+      tuningSet: Uniform5qps
+      objectBundle:
+        - basename: deployment
+          objectTemplatePath: deployment.yaml
+          templateFillMap:
+            Replicas: {{$podsPerNamespace}}
+            Group: nichotr
+            CpuRequest: {{$POD_CPU}}m
+            MemoryRequest: {{$POD_MEMORY}}M
+
+- name: Collecting pod measurements
+- measurements:
+    - Identifier: WaitingForPods
+      Method: WaitForRunningPods
+      Params:
+        labelSelector: group = nichotr
+        desiredPodCount: {{$totalPods}}
+        timeout: 1h
+
+- name: Timing Separation For Scale Up
+- measurements:
+    - Identifier: ScaleUpTimer
+      Method: Timer
+      Params:
+        action: stop
+        label: scale_up_time
+    - Identifier: ScaleDownTimer
+      Method: Timer
+      Params:
+        action: start
+        label: scale_down_time
+
+- name: Deleting saturation pods
+  phases:
+    - namespaceRange:
+        min: 1
+        max: {{$namespaces}}
+      replicasPerNamespace: 0
+      tuningSet: Uniform5qps
+      objectBundle:
+        - basename: deployment
+          objectTemplatePath: deployment.yaml
+
+- name: Waiting for Deletion
+- measurements:
+    - Identifier: WaitingForPods
+      Method: WaitForRunningPods
+      Params:
+        labelSelector: group = nichotr
+        desiredPodCount: 0
+        timeout: 1h
+
+- name: Timing Separation for Scale Down
+- measurements:
+    - Identifier: ScaleDownTimer
+      Method: Timer
+      Params:
+        action: stop
+        label: scale_down_time
+    - Identifier: PodStartupLatency
+      Method: PodStartupLatency
+      Params:
+        action: gather
+    - Identifier: ResourceUsageSummary
+      Method: ResourceUsageSummary
+      Params:
+        action: gather
+    - Identifier: SchedulingThroughput
+      Method: SchedulingThroughput
```

Comment:
Will these measurements be impacted by the fact that we've scaled down already? Is there any order of operations required?

Reply:
I believe the measurements keep track of each step along the way, still looking into it. If not, it's as simple as adding another start and stop somewhere along the way.